src/stella_vslam/solve/essential_solver.cc

```diff
@@ -48,28 +49,41 @@ void essential_solver::find_via_ransac(const unsigned int max_num_iter, const bo
             min_set_bearings_2.at(i) = bearings_2_.at(matches_12_.at(idx).second);
         }
 
-        // 2-2. Compute an essential matrix
-        E_21_in_sac = compute_E_21(min_set_bearings_1, min_set_bearings_2);
-
-        // 2-3. Check inliers and compute a cost
-        float cost_in_sac;
-        unsigned int num_inliers = check_inliers(E_21_in_sac, is_inlier_match_in_sac, cost_in_sac);
+        // 2-2. Compute candidate essential matrices with the minimal solver
+        std::vector<Mat33_t> E_mats;
+        if (min_set_size == 5) {
```

Comment:
I would like to add an assert.

```suggestion
        assert(min_set_size == 5 || min_set_size >= 8);
        if (min_set_size == 5) {
```

or

```suggestion
        assert(min_set_size >= 5);
        if (min_set_size < 8) {
```

Reply:
ah great catch! :+1: I like the second one best, since compute_E_21_minimal will work with 5+ points. so using it if there are < 8 points makes sense.